// @rollpost/contract/dist/chunk-6UE6XDGS.js
// src/common/schemas.ts
import { z } from "zod";
var UUIDSchema = z.string().uuid();
var ApiVersionSchema = z.literal("v1");
var ApiErrorSchema = z.object({
  error: z.string().min(1),
  code: z.string().min(1).optional(),
  details: z.unknown().optional()
});
var ERROR_CODES = {
  UNAUTHORIZED: "UNAUTHORIZED",
  FORBIDDEN: "FORBIDDEN",
  NOT_FOUND: "NOT_FOUND",
  VALIDATION_ERROR: "VALIDATION_ERROR",
  CONFLICT: "CONFLICT",
  INTERNAL_ERROR: "INTERNAL_ERROR"
};
var errorResponseSchema = z.object({
  error: z.string().min(1),
  code: z.nativeEnum(ERROR_CODES),
  field: z.string().optional()
});
var cursorQuerySchema = z.object({
  cursor: z.string().optional(),
  limit: z.coerce.number().int().min(1).max(100).default(50)
});
var paginatedResponseSchema = (itemSchema) => z.object({
  items: z.array(itemSchema),
  nextCursor: z.string().nullable(),
  hasMore: z.boolean()
});

export {
  UUIDSchema,
  ApiVersionSchema,
  ApiErrorSchema,
  ERROR_CODES,
  errorResponseSchema,
  cursorQuerySchema,
  paginatedResponseSchema
};
