// @rollpost/contract/dist/index.js
import "./chunk-MBMZM54F.js";
import {
  ApiErrorSchema,
  ApiVersionSchema,
  ERROR_CODES,
  UUIDSchema,
  cursorQuerySchema,
  errorResponseSchema,
  paginatedResponseSchema
} from "./chunk-6UE6XDGS.js";

// src/auth.ts
import { z } from "zod";
var signUpSchema = z.object({
  email: z.string().email(),
  password: z.string().min(8)
});
var signInSchema = z.object({
  email: z.string().email(),
  password: z.string().min(1)
});
var authResponseSchema = z.object({
  sessionToken: z.string(),
  userId: z.string().uuid()
});
var subscriptionStatusSchema = z.enum([
  "active",
  "inactive",
  "trialing"
]);
var currentUserSchema = z.object({
  id: z.string().uuid(),
  email: z.string().email(),
  avatarUrl: z.string().url().nullable(),
  subscriptionStatus: subscriptionStatusSchema,
  activeGameCount: z.number().int().min(0),
  emailVerified: z.boolean(),
  createdAt: z.string().datetime()
});

// src/games.ts
import { z as z2 } from "zod";
var gameStatusSchema = z2.enum(["active", "archived", "paused"]);
var storageStatusSchema = z2.enum(["hot", "cold"]);
var createGameSchema = z2.object({
  title: z2.string().min(1).max(100),
  description: z2.string().max(1e3).optional(),
  system: z2.string().min(1).max(100),
  maxPlayers: z2.number().int().min(1).max(8).default(8)
});
var gameSchema = z2.object({
  id: z2.string().uuid(),
  title: z2.string(),
  description: z2.string().nullable(),
  system: z2.string(),
  maxPlayers: z2.number().int(),
  status: gameStatusSchema,
  storageStatus: storageStatusSchema,
  dmUserId: z2.string().uuid(),
  createdAt: z2.string().datetime(),
  updatedAt: z2.string().datetime()
});
var gameParamsSchema = z2.object({
  gameId: z2.string().uuid()
});

// src/billing.ts
import { z as z3 } from "zod";
var createCheckoutSchema = z3.object({
  userId: z3.string().uuid(),
  successUrl: z3.string().url(),
  cancelUrl: z3.string().url()
});
var checkoutSessionSchema = z3.object({
  sessionId: z3.string(),
  url: z3.string().url()
});
var stripeWebhookMetaSchema = z3.object({
  type: z3.string(),
  userId: z3.string().uuid().optional()
});

// src/invites.ts
import { z as z4 } from "zod";
var generateInviteSchema = z4.object({
  expiresInHours: z4.number().int().min(1).max(168).optional(),
  // up to 7 days; omit = no expiry
  note: z4.string().max(500).optional()
});
var inviteSchema = z4.object({
  id: z4.string().uuid(),
  token: z4.string(),
  gameId: z4.string().uuid(),
  createdByUserId: z4.string().uuid(),
  used: z4.boolean(),
  expiresAt: z4.string().datetime().nullable(),
  createdAt: z4.string().datetime()
});
var inviteTokenParamsSchema = z4.object({
  token: z4.string().min(1)
});
var invitePreviewSchema = z4.discriminatedUnion("isValid", [
  z4.object({
    isValid: z4.literal(true),
    gameId: z4.string().uuid(),
    gameTitle: z4.string(),
    gameSystem: z4.string(),
    dmDisplayName: z4.string(),
    currentPlayerCount: z4.number().int(),
    maxPlayers: z4.number().int(),
    expiresAt: z4.string().datetime().nullable()
  }),
  z4.object({
    isValid: z4.literal(false),
    reason: z4.enum(["expired", "already_used", "not_found"])
  })
]);
var characterFieldsSchema = z4.object({
  characterName: z4.string().min(1).max(100),
  characterDescription: z4.string().max(2e3).optional()
});
var acceptInviteSchema = z4.discriminatedUnion("hasSession", [
  // Existing / already-logged-in user
  z4.object({
    hasSession: z4.literal(true)
  }).merge(characterFieldsSchema),
  // New account — create user + character atomically
  z4.object({
    hasSession: z4.literal(false),
    email: z4.string().email(),
    password: z4.string().min(8)
  }).merge(characterFieldsSchema)
]);
var acceptInviteResponseSchema = z4.object({
  memberId: z4.string().uuid(),
  characterId: z4.string().uuid(),
  gameId: z4.string().uuid(),
  isNewAccount: z4.boolean(),
  sessionToken: z4.string().optional()
  // only present when isNewAccount=true
});
export {
  ApiErrorSchema,
  ApiVersionSchema,
  ERROR_CODES,
  UUIDSchema,
  acceptInviteResponseSchema,
  acceptInviteSchema,
  authResponseSchema,
  checkoutSessionSchema,
  createCheckoutSchema,
  createGameSchema,
  currentUserSchema,
  cursorQuerySchema,
  errorResponseSchema,
  gameParamsSchema,
  gameSchema,
  gameStatusSchema,
  generateInviteSchema,
  invitePreviewSchema,
  inviteSchema,
  inviteTokenParamsSchema,
  paginatedResponseSchema,
  signInSchema,
  signUpSchema,
  storageStatusSchema,
  stripeWebhookMetaSchema,
  subscriptionStatusSchema
};
