// @rollpost/contract/dist/index.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  ApiErrorSchema: () => ApiErrorSchema,
  ApiVersionSchema: () => ApiVersionSchema,
  ERROR_CODES: () => ERROR_CODES,
  UUIDSchema: () => UUIDSchema,
  acceptInviteResponseSchema: () => acceptInviteResponseSchema,
  acceptInviteSchema: () => acceptInviteSchema,
  authResponseSchema: () => authResponseSchema,
  checkoutSessionSchema: () => checkoutSessionSchema,
  createCheckoutSchema: () => createCheckoutSchema,
  createGameSchema: () => createGameSchema,
  currentUserSchema: () => currentUserSchema,
  cursorQuerySchema: () => cursorQuerySchema,
  errorResponseSchema: () => errorResponseSchema,
  gameParamsSchema: () => gameParamsSchema,
  gameSchema: () => gameSchema,
  gameStatusSchema: () => gameStatusSchema,
  generateInviteSchema: () => generateInviteSchema,
  invitePreviewSchema: () => invitePreviewSchema,
  inviteSchema: () => inviteSchema,
  inviteTokenParamsSchema: () => inviteTokenParamsSchema,
  paginatedResponseSchema: () => paginatedResponseSchema,
  signInSchema: () => signInSchema,
  signUpSchema: () => signUpSchema,
  storageStatusSchema: () => storageStatusSchema,
  stripeWebhookMetaSchema: () => stripeWebhookMetaSchema,
  subscriptionStatusSchema: () => subscriptionStatusSchema
});
module.exports = __toCommonJS(index_exports);

// src/common/schemas.ts
var import_zod = require("zod");
var UUIDSchema = import_zod.z.string().uuid();
var ApiVersionSchema = import_zod.z.literal("v1");
var ApiErrorSchema = import_zod.z.object({
  error: import_zod.z.string().min(1),
  code: import_zod.z.string().min(1).optional(),
  details: import_zod.z.unknown().optional()
});
var ERROR_CODES = {
  UNAUTHORIZED: "UNAUTHORIZED",
  FORBIDDEN: "FORBIDDEN",
  NOT_FOUND: "NOT_FOUND",
  VALIDATION_ERROR: "VALIDATION_ERROR",
  CONFLICT: "CONFLICT",
  INTERNAL_ERROR: "INTERNAL_ERROR"
};
var errorResponseSchema = import_zod.z.object({
  error: import_zod.z.string().min(1),
  code: import_zod.z.nativeEnum(ERROR_CODES),
  field: import_zod.z.string().optional()
});
var cursorQuerySchema = import_zod.z.object({
  cursor: import_zod.z.string().optional(),
  limit: import_zod.z.coerce.number().int().min(1).max(100).default(50)
});
var paginatedResponseSchema = (itemSchema) => import_zod.z.object({
  items: import_zod.z.array(itemSchema),
  nextCursor: import_zod.z.string().nullable(),
  hasMore: import_zod.z.boolean()
});

// src/auth.ts
var import_zod2 = require("zod");
var signUpSchema = import_zod2.z.object({
  email: import_zod2.z.string().email(),
  password: import_zod2.z.string().min(8)
});
var signInSchema = import_zod2.z.object({
  email: import_zod2.z.string().email(),
  password: import_zod2.z.string().min(1)
});
var authResponseSchema = import_zod2.z.object({
  sessionToken: import_zod2.z.string(),
  userId: import_zod2.z.string().uuid()
});
var subscriptionStatusSchema = import_zod2.z.enum([
  "active",
  "inactive",
  "trialing"
]);
var currentUserSchema = import_zod2.z.object({
  id: import_zod2.z.string().uuid(),
  email: import_zod2.z.string().email(),
  avatarUrl: import_zod2.z.string().url().nullable(),
  subscriptionStatus: subscriptionStatusSchema,
  activeGameCount: import_zod2.z.number().int().min(0),
  emailVerified: import_zod2.z.boolean(),
  createdAt: import_zod2.z.string().datetime()
});

// src/games.ts
var import_zod3 = require("zod");
var gameStatusSchema = import_zod3.z.enum(["active", "archived", "paused"]);
var storageStatusSchema = import_zod3.z.enum(["hot", "cold"]);
var createGameSchema = import_zod3.z.object({
  title: import_zod3.z.string().min(1).max(100),
  description: import_zod3.z.string().max(1e3).optional(),
  system: import_zod3.z.string().min(1).max(100),
  maxPlayers: import_zod3.z.number().int().min(1).max(8).default(8)
});
var gameSchema = import_zod3.z.object({
  id: import_zod3.z.string().uuid(),
  title: import_zod3.z.string(),
  description: import_zod3.z.string().nullable(),
  system: import_zod3.z.string(),
  maxPlayers: import_zod3.z.number().int(),
  status: gameStatusSchema,
  storageStatus: storageStatusSchema,
  dmUserId: import_zod3.z.string().uuid(),
  createdAt: import_zod3.z.string().datetime(),
  updatedAt: import_zod3.z.string().datetime()
});
var gameParamsSchema = import_zod3.z.object({
  gameId: import_zod3.z.string().uuid()
});

// src/billing.ts
var import_zod4 = require("zod");
var createCheckoutSchema = import_zod4.z.object({
  userId: import_zod4.z.string().uuid(),
  successUrl: import_zod4.z.string().url(),
  cancelUrl: import_zod4.z.string().url()
});
var checkoutSessionSchema = import_zod4.z.object({
  sessionId: import_zod4.z.string(),
  url: import_zod4.z.string().url()
});
var stripeWebhookMetaSchema = import_zod4.z.object({
  type: import_zod4.z.string(),
  userId: import_zod4.z.string().uuid().optional()
});

// src/invites.ts
var import_zod5 = require("zod");
var generateInviteSchema = import_zod5.z.object({
  expiresInHours: import_zod5.z.number().int().min(1).max(168).optional(),
  // up to 7 days; omit = no expiry
  note: import_zod5.z.string().max(500).optional()
});
var inviteSchema = import_zod5.z.object({
  id: import_zod5.z.string().uuid(),
  token: import_zod5.z.string(),
  gameId: import_zod5.z.string().uuid(),
  createdByUserId: import_zod5.z.string().uuid(),
  used: import_zod5.z.boolean(),
  expiresAt: import_zod5.z.string().datetime().nullable(),
  createdAt: import_zod5.z.string().datetime()
});
var inviteTokenParamsSchema = import_zod5.z.object({
  token: import_zod5.z.string().min(1)
});
var invitePreviewSchema = import_zod5.z.discriminatedUnion("isValid", [
  import_zod5.z.object({
    isValid: import_zod5.z.literal(true),
    gameId: import_zod5.z.string().uuid(),
    gameTitle: import_zod5.z.string(),
    gameSystem: import_zod5.z.string(),
    dmDisplayName: import_zod5.z.string(),
    currentPlayerCount: import_zod5.z.number().int(),
    maxPlayers: import_zod5.z.number().int(),
    expiresAt: import_zod5.z.string().datetime().nullable()
  }),
  import_zod5.z.object({
    isValid: import_zod5.z.literal(false),
    reason: import_zod5.z.enum(["expired", "already_used", "not_found"])
  })
]);
var characterFieldsSchema = import_zod5.z.object({
  characterName: import_zod5.z.string().min(1).max(100),
  characterDescription: import_zod5.z.string().max(2e3).optional()
});
var acceptInviteSchema = import_zod5.z.discriminatedUnion("hasSession", [
  // Existing / already-logged-in user
  import_zod5.z.object({
    hasSession: import_zod5.z.literal(true)
  }).merge(characterFieldsSchema),
  // New account — create user + character atomically
  import_zod5.z.object({
    hasSession: import_zod5.z.literal(false),
    email: import_zod5.z.string().email(),
    password: import_zod5.z.string().min(8)
  }).merge(characterFieldsSchema)
]);
var acceptInviteResponseSchema = import_zod5.z.object({
  memberId: import_zod5.z.string().uuid(),
  characterId: import_zod5.z.string().uuid(),
  gameId: import_zod5.z.string().uuid(),
  isNewAccount: import_zod5.z.boolean(),
  sessionToken: import_zod5.z.string().optional()
  // only present when isNewAccount=true
});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ApiErrorSchema,
  ApiVersionSchema,
  ERROR_CODES,
  UUIDSchema,
  acceptInviteResponseSchema,
  acceptInviteSchema,
  authResponseSchema,
  checkoutSessionSchema,
  createCheckoutSchema,
  createGameSchema,
  currentUserSchema,
  cursorQuerySchema,
  errorResponseSchema,
  gameParamsSchema,
  gameSchema,
  gameStatusSchema,
  generateInviteSchema,
  invitePreviewSchema,
  inviteSchema,
  inviteTokenParamsSchema,
  paginatedResponseSchema,
  signInSchema,
  signUpSchema,
  storageStatusSchema,
  stripeWebhookMetaSchema,
  subscriptionStatusSchema
});
