// @rollpost/contract/dist/common/index.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/common/index.ts
var common_exports = {};
__export(common_exports, {
  ApiErrorSchema: () => ApiErrorSchema,
  ApiVersionSchema: () => ApiVersionSchema,
  ERROR_CODES: () => ERROR_CODES,
  UUIDSchema: () => UUIDSchema,
  cursorQuerySchema: () => cursorQuerySchema,
  errorResponseSchema: () => errorResponseSchema,
  paginatedResponseSchema: () => paginatedResponseSchema
});
module.exports = __toCommonJS(common_exports);

// src/common/schemas.ts
var import_zod = require("zod");
var UUIDSchema = import_zod.z.string().uuid();
var ApiVersionSchema = import_zod.z.literal("v1");
var ApiErrorSchema = import_zod.z.object({
  error: import_zod.z.string().min(1),
  code: import_zod.z.string().min(1).optional(),
  details: import_zod.z.unknown().optional()
});
var ERROR_CODES = {
  UNAUTHORIZED: "UNAUTHORIZED",
  FORBIDDEN: "FORBIDDEN",
  NOT_FOUND: "NOT_FOUND",
  VALIDATION_ERROR: "VALIDATION_ERROR",
  CONFLICT: "CONFLICT",
  INTERNAL_ERROR: "INTERNAL_ERROR"
};
var errorResponseSchema = import_zod.z.object({
  error: import_zod.z.string().min(1),
  code: import_zod.z.nativeEnum(ERROR_CODES),
  field: import_zod.z.string().optional()
});
var cursorQuerySchema = import_zod.z.object({
  cursor: import_zod.z.string().optional(),
  limit: import_zod.z.coerce.number().int().min(1).max(100).default(50)
});
var paginatedResponseSchema = (itemSchema) => import_zod.z.object({
  items: import_zod.z.array(itemSchema),
  nextCursor: import_zod.z.string().nullable(),
  hasMore: import_zod.z.boolean()
});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ApiErrorSchema,
  ApiVersionSchema,
  ERROR_CODES,
  UUIDSchema,
  cursorQuerySchema,
  errorResponseSchema,
  paginatedResponseSchema
});
